ui/src/components/ComponentTypeSelect/ComponentTypeSelect.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { PropsWithChildren } from 'react';
+
+import Spinner from '@atlaskit/spinner';
+import Select, { ActionMeta } from '@atlaskit/select';
+import InlineMessage from '@atlaskit/inline-message';
+import { Centered } from './styles';
+import { CompassComponentTypeId, CompassComponentTypeOption } from '../../services/types';
+import { FormatOptionLabel } from '../FormatOptionLabel';
+import { DEFAULT_COMPONENT_TYPE_ID } from '../../constants';
+import { getComponentTypeOption } from '../utils';
+
+type ComponentTypeSelector = {
+  loading: boolean;
+  dropdownId: string;
+  componentTypes?: CompassComponentTypeId[];
+  isDisabled?: boolean | undefined;
+  onChange?:
+    | ((value: CompassComponentTypeOption | null, action: ActionMeta<CompassComponentTypeOption>) => void)
+    | undefined;
+  selectedOption?: CompassComponentTypeOption | null;
+};
+
+const isComponentTypesEmpty = (componentTypes?: CompassComponentTypeId[]): boolean => {
+  return !componentTypes || componentTypes.length === 0;
+};
+
+const getComponentTypeIdItems = (componentTypes?: CompassComponentTypeId[]): CompassComponentTypeOption[] => {
+  if (!isComponentTypesEmpty(componentTypes)) {
+    return (
+      componentTypes?.map((componentTypeId) => {
+        return getComponentTypeOption(componentTypeId?.id);
+      }) ?? []
+    );
+  }
+  return [];
+};
+
+export const ComponentTypeSelect: React.FC<ComponentTypeSelector> = (
+  props: PropsWithChildren<ComponentTypeSelector>,
+) => {
+  const { componentTypes, loading, dropdownId, isDisabled, onChange, selectedOption } = props;
+
+  if (loading) {
+    return <Spinner size={'small'} />;
+  }
+
+  if (isComponentTypesEmpty(componentTypes)) {
+    return (
+      <Centered>
+        <InlineMessage appearance='error' iconLabel='Error loading component types'>
```

Comment:
Do we have a ui test for this?

Reply:
Not currently @rharibhakti 
However, I will create one now. 
Thank you for pointing this out.